### PR TITLE
Deserialize long values without throwing an exception when type isn't specified

### PR DESCRIPTION
**Pull request by mduffin95**
Beans containing maps with `Object`s throw an exception when deserializing a numeric value that lies outside of the integer min/max range. 

## Summary by CodeRabbit

- **New Features**
	- Enhanced the `SimpleJson` class with a new property, `objectInMap`, allowing for a broader range of object types.
	- Added new methods for accessing the `objectInMap` property, improving usability.

- **Bug Fixes**
	- Improved handling of numeric values during parsing to prevent errors when values exceed integer limits, enhancing robustness.

### CODE REVIEW

File: src/test/java/org/joda/beans/sample/SimpleJson.java [hunk 1]
```diff
@@ -99,1564 +99,1618 @@ public final class SimpleJson implements ImmutableBean, Serializable {
     private final ImmutableMap<String, List<String>> listInMap;
     @PropertyDefinition
     private final ImmutableMap<String, List<Integer>> listNumericInMap;
+    @PropertyDefinition
+    private final ImmutableMap<String, Object> objectInMap;
+
+  //------------------------- AUTOGENERATED START -------------------------
```
Comment (mduffin95) [inline src/test/java/org/joda/beans/sample/SimpleJson.java:105]: I believe this large diff must be due to using a later version of JodaBeans.

Comment (jodastephen) [reply src/test/java/org/joda/beans/sample/SimpleJson.java:105]: The problem is that it is generated with 2 space indent when it needs to be 4 space. You'll need to generate manually or adjust the plugin settings.

**Review by jodastephen — CHANGES_REQUESTED** [review]

**Comment by jodastephen** [comment]
FWIW, part of me thinks the default should always be Long, not Integer/Long depending on the size of the value. Obviously incompatible, but more sensible.

File: src/test/java/org/joda/beans/sample/SimpleJson.java (hunk 1 above)
Comment (mduffin95) [reply src/test/java/org/joda/beans/sample/SimpleJson.java:105]: Ah yes hadn't noticed that

File: src/main/java/org/joda/beans/ser/bin/AbstractBinReader.java [hunk 2]
```diff
@@ -322,11 +322,13 @@ Object parseSimple(int typeByte, Class<?> type) throws Exception {
...
                 return Float.valueOf((float) value);
 
-            } else {
+            } else if (type == Integer.class || type == int.class) {
                 if (value < Integer.MIN_VALUE || value > Integer.MAX_VALUE) {
                     throw new IllegalArgumentException("Invalid binary data: Expected int, but was " + value);
                 }
                 return Integer.valueOf((int) value);
+            } else {
+                return Long.valueOf(value);
```
Comment (mduffin95) [inline src/main/java/org/joda/beans/ser/bin/AbstractBinReader.java:331]: I've changed the bin reader too for consistency, even though I don't think it suffers from the same issue as we don't ever serialize to binary format without type information.

File: src/main/java/org/joda/beans/ser/bin/AbstractBinReader.java [hunk 3]
```diff
@@ -323,10 +325,7 @@ Object parseSimple(int typeByte, Class<?> type) throws Exception {
                 return Float.valueOf((float) value);
 
             } else {
-                if (value < Integer.MIN_VALUE || value > Integer.MAX_VALUE) {
-                    throw new IllegalArgumentException("Invalid binary data: Expected int, but was " + value);
-                }
-                return Integer.valueOf((int) value);
+                return Long.valueOf(value);
```
Comment (mduffin95) [inline src/main/java/org/joda/beans/ser/bin/AbstractBinReader.java:328]: I've made the change here too for consistency but the binary reader deduces the type correctly.

**Review by jodastephen — APPROVED** [review]

**Comment by jodastephen** [comment]
Test failures

**Comment by mduffin95** [comment]
Hmm, I hadn't appreciated that defaulting to Long would affect serialisation too. Now we'll need ot include "@type: Integer" if a JSON number is a `Long`. I might just go back to the initial basic fix and perhaps we can look into what defaulting to Long might entail later.

**Review by jodastephen — APPROVED** [review]

**Pull request merged**